api/src/main/java/io/opentelemetry/context/propagation/DefaultContextPropagators.java

```diff
@@ -0,0 +1,158 @@
+ * ContextPropagators}.
+ *
+ * <p>All the registered propagators are stored internally as a simple list, and are invoked
+ * synchronically upon injection and extraction.
+ *
+ * @since 0.3.0
+ */
+public final class DefaultContextPropagators implements ContextPropagators {
+  private final HttpTextFormat textFormat;
+
+  @Override
+  public HttpTextFormat getHttpTextFormat() {
+    return textFormat;
+  }
+
+  /**
+   * Returns a {@link DefaultContextPropagators.Builder} to create a new {@link ContextPropagators}
+   * object.
+   *
+   * @return a {@link DefaultContextPropagators.Builder}.
+   * @since 0.3.0
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  private DefaultContextPropagators(HttpTextFormat textFormat) {
+    this.textFormat = textFormat;
+  }
+
+  /**
+   * {@link Builder} is used to construct a new {@code ContextPropagators} object with the specified
+   * propagators.
+   *
+   * <p>This is a example of a {@code ContextPropagators} object being created:
+   *
+   * <pre>{@code
+   * ContextPropagators propagators = DefaultContextPropagators.builder()
+   *     .addHttpTextFormat(new HttpTraceContext())
+   *     .addHttpTextFormat(new HttpCorrelationContext())
+   *     .addHttpTextFormat(new MyCustomContextPropagator())
+   *     .build();
+   * }</pre>
+   *
+   * @since 0.3.0
+   */
+  public static final class Builder {
+    List<HttpTextFormat> textPropagators = new ArrayList<>();
+
+    /**
+     * Adds a {@link HttpTextFormat} propagator.
+     *
+     * <p>One propagator per concern (traces, correlations, etc) should be added if this format is
+     * supported.
+     *
+     * @param textFormat the propagator to be added.
+     * @return this.
+     * @throws NullPointerException if {@code textFormat} is {@code null}.
+     * @since 0.3.0
+     */
+    public Builder addHttpTextFormat(HttpTextFormat textFormat) {
+      Utils.checkNotNull(textFormat, "textFormat");
+      textPropagators.add(textFormat);
+      return this;
+    }
+
+    /**
+     * Builds a new {@code ContextPropagators} with the specified propagators.
+     *
+     * @return the newly created {@code ContextPropagators} instance.
+     * @since 0.3.0
+     */
+    public ContextPropagators build() {
+      if (textPropagators.isEmpty()) {
+        return new DefaultContextPropagators(NoopHttpTextFormat.INSTANCE);
+      }
+
+      return new DefaultContextPropagators(new MultiHttpTextFormat(textPropagators));
+    }
+  }
+
+  private static final class MultiHttpTextFormat implements HttpTextFormat {
+    private final HttpTextFormat[] textPropagators;
+
+    private MultiHttpTextFormat(List<HttpTextFormat> textPropagators) {
+      this.textPropagators = new HttpTextFormat[textPropagators.size()];
+      textPropagators.toArray(this.textPropagators);
+    }
+
+    @Override
+    public List<String> fields() {
+      List<String> fields = new ArrayList<>();
```

Comment:
Caching sounds great. Throwing an exception in case of repeated fields, not sure. I'd rather have them as duplicated entries in the returned `List` :)

Reply:
>  the fields method has to always return the same set of fields.

This we can do, yes ;)

Reply:
Caching for now - lets figure out later on whether we should return them all or de-duplicate them.